File: ayudantia8_2019/src/ayudantia8_2019/ListaJuegos.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package ayudantia8_2019;

/**
 *
 * @author Juan Pablo Martinez
 */
public class ListaJuegos {
    private int cant;
    private int max;
    private Juego [] lj;

    public ListaJuegos(int max) {
        this.cant = 0;
        this.max = max;
        this.lj = new Juego[max];
    }

    public int getCant() {
        return cant;
    }

    public void setCant(int cant) {
        this.cant = cant;
    }

    public int getMax() {
        return max;
    }

    public void setMax(int max) {
        this.max = max;
    }

    public Juego getJuego(int i) {
        return lj[i];
    }

    public void setLj(Juego[] lj) {
        this.lj = lj;
    }
    
    public boolean ingresarJuego(Juego j){
        if(cant < max){
            lj[cant] =j;
            cant++;
            return true;
        }else{
            return false;
        }
    }
    
    public Juego buscarJuego(String nombre){
        int i;
        for(i = 0;i<cant;i++){
            if(lj[i].getNombre().equalsIgnoreCase(nombre)){
                break;
            }
        }
        if(i == cant){
            return null;
        }else{
            return lj[i];
        }
    }
    
}

File: Taller1ED_2018/ListaJuegos.cpp
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/* 
 * File:   ListaJuegos.cpp
 * Author: Casa
 * 
 * Created on 27 de abril de 2018, 13:52
 */

#include "ListaJuegos.h"
#include <cstdlib>
#include <string.h>
#include <iostream>
using namespace std;

ListaJuegos::ListaJuegos() {
    this->first=NULL;
    this->last=NULL;
}

NodoJuego* ListaJuegos::getFirst() {
    return first;
}

NodoJuego* ListaJuegos::getLast() {
    return last;
}

void ListaJuegos::setFirst(NodoJuego*first) {
    this->first=first;
}

void ListaJuegos::setLast(NodoJuego*last) {
    this->last=last;
}

void ListaJuegos::ingresarJuego(Juego* J) {
    NodoJuego *nuevo = new NodoJuego(J);
    if(first==NULL) {
        first=nuevo;
        last=nuevo;
    }else{
        nuevo->setNext(first);
        first->setPrev(nuevo);
        nuevo->setPrev(NULL);
        first=nuevo;
    }

}

Juego* ListaJuegos::buscarJuego(string codigo) {
    NodoJuego *current = first;
    while(current != NULL && current->getJuego()->getCodigo()!= codigo){
        current = current->getNext();
    }   
    if(current != NULL) {
        return current->getJuego();
    }else{
        return NULL;
    }

}

bool ListaJuegos::eliminarJuego(string codigo) {
    NodoJuego *current = first;
    while(current!= NULL && current->getJuego()->getCodigo() != codigo) {
        current = current->getPrev();
    }
    if(current != NULL){
        if(current == first) {
            NodoJuego *eliminado = first;
            first = first->getNext();
            delete eliminado;
        }else{
            current->getPrev()->setNext(current->getNext());
        }
        if(current == last) {
            NodoJuego *eliminado = last;
            last = last->getPrev();
            delete eliminado;
        }else{
            current->getNext()->setPrev(current->getPrev());
        }
        return true;
    }else{
        return false;
    }
}



ListaJuegos::~ListaJuegos() {
}

